ledger/participant-integration-api/src/main/scala/platform/store/backend/postgresql/PostgresStorageBackend.scala

```diff
@@ -86,6 +88,37 @@ private[backend] object PostgresStorageBackend
 
   override val duplicateKeyError: String = "duplicate key"
 
+  def getPostgresVersion(connection: Connection): Option[(Int, Int, Int)] = {
+    val version = SQL"SHOW server_version".as(get[String](1).single)(connection)
+    val versionPattern = """(\d)[.](\d)[.](\d)""".r
+    version match {
+      case versionPattern(major, minor, patch) => Some(major.toInt, minor.toInt, patch.toInt)
+      case _ => None
+    }
+  }
+
+  override def checkCompatibility(
+      connection: Connection
+  )(implicit loggingContext: LoggingContext): Unit = {
+    getPostgresVersion(connection) match {
+      case Some((major, minor, patch)) =>
+        if (major.toInt < 10) {
+          logger.error(
+            "Deprecated Postgres version. " +
+              s"Found Postgres version $major.$minor.$patch., minimum required Postgres version is 10. " +
+              "This application will continue running but is at risk of data loss, as Postgres < 10 does not support crash-fault tolerant hash indices. " +
+              "Please upgrade your Postgres database to version 10 or later to fix this issue. " +
+              "In the future, this deprecation warning may be upgraded to a fatal error."
+          )
+        }
+      case None =>
+        logger.warn(
+          s"Could not determine the version of the Postgres database. Please verify that this application is compatible with this Postgres version."
```

Comment:
if we fail with an error if we detect a wrong version, then perhaps we should fail with an error if version cannot be read?
BTW where are the requirements coming from wtr? (why error log in contrast to error log and fail?)

Reply:
This PR only "deprecates" old Postgres versions, it doesn't yet force users to upgrade. Later we can change both log outputs to throwing an error. which would force users to upgrade.